benchadapt/python/benchadapt/_machine_info.py

```diff
@@ -76,25 +79,46 @@ class GitParseWarning(RuntimeWarning):
     pass
 
 
-def github_info():
-    """Get github metadata from environment variables"""
-    repo = os.environ.get("CONBENCH_PROJECT_REPOSITORY")
-    pr_number = os.environ.get("CONBENCH_PROJECT_PR_NUMBER") or os.environ.get(
-        "BENCHMARKABLE_PR_NUMBER"
-    )
-    commit = os.environ.get("CONBENCH_PROJECT_COMMIT")
+def gh_commit_info_from_env() -> Dict[str, str]:
+    """
+    Attempt to read a specific set of environment variables expected to carry
+    GitHub-flavored information about the commit/checkout state of the
+    benchmarked repository.
+
+    The returned dictionary is guaranteed to have both keys and values of type
+    str, all of length greater zero.
 
-    gh_info = {
-        "commit": commit,
-        "repository": repo,
-        # "branch": None,
-        "pr_number": pr_number,
+    The returned dictionary may be empty.
+    """
+
+    varmap = {
+        "CONBENCH_PROJECT_REPOSITORY": "repository",
+        "CONBENCH_PROJECT_COMMIT": "commit",
+        "BENCHMARKABLE_PR_NUMBER": "pr_number",
+        "CONBENCH_PROJECT_PR_NUMBER": "pr_number",  # later entry takes precedence
     }
 
-    return gh_info
+    result = {}
+
+    for evarname, dictkey in varmap.items():
+        val = os.environ.get(evarname)
+        if val is not None:
+            stripval = val.strip()
 
+            if len(stripval) > 0:
+                result[dictkey] = stripval
+                continue
 
-def detect_github_info():
+            log.warning(
```

Comment:
I don't know if this logger will find a root logger to bubble messages into. But if that's not yet the case, then these additional log messages will not break anything, and they will find a root logger in the future, when we consolidate logging.